test-crates/duchess-java-tests/tests/ui/native_fn_callable_from_java.rs

```diff
@@ -0,0 +1,21 @@
+//@compile-flags: --crate-type cdylib
+//@check-pass
+//@rustc-env: CLASSPATH=../target/tests/java_ui
```

Comment:
Hmm, I like this capability, I don't know if I like mixing it in with the other `ui` tests. What about making a separate directory where these compile-flags and rustc-env options are the default? Something like:

* `src/test/rs-to-java`
* `src/test/rs-from-java` <-- the new one
* `src/test/js-to-rs`

or even just put the `.rs` files into the `src/test/js-to-rs` directory as well?

Reply:
I like your 3 directory proposal, I'll go with that, but as far as naming and structure, what about...

* ```src/test/java-to-rust/java```
* ```src/test/java-to-rust/rust_libraries```
* ```src/test/rs-to-java```